ext/FlyThroughPathsMakieExt.jl

```diff
@@ -11,11 +11,98 @@ FlyThroughPaths.capture_view(scene::Scene) = capture_view(cameracontrols(scene))
-    cameracontrols(scene).fov[] = view.fov[]
+    # Extract the camera controls from the Scene
+    cam = Makie.cameracontrols(scene)
+    @assert cam isa Makie.Camera3D "`cameracontrols(scene)` must be a `Camera3D`, we don't support any other camera.  Got $(typeof(cam))."
+    # Set the appropriate fields
+    cam.eyeposition[] = view.eyeposition
+    cam.lookat[] = view.lookat
+    cam.upvector[] = view.upvector
+    cam.fov[] = view.fov
+    # Update the camera using the new controls
+    Makie.update_cam!(scene, cam)
     return scene
 end
-
 FlyThroughPaths.set_view!(axis::Makie.AbstractAxis, view::ViewState) = set_view!(axis.scene, view)
 
+function Makie.record(fig::Makie.FigureLike, file::String, path::Path; framerate = 24, kwargs...)
+    tend = FlyThroughPaths.duration(path)
+    trange = LinRange(0, tend, round(Int, tend / framerate))
+    iterator = path.(trange)
+    return Makie.record(fig, file, iterator; framerate, kwargs...)
+end
+
+# Define the recipe
+import FlyThroughPaths: plotcamerapath, plotcamerapath!
+@recipe(PlotCameraPath, path, time) do scene
+    Attributes(
+        colormap = Makie.inherit(scene, :colormap, :plasma),
+        color = Makie.inherit(scene, :color, :black),
+        linewidth = Makie.inherit(scene, :linewidth, 1.0),
+        linestyle = Makie.inherit(scene, :linestyle, :solid),
+        camera_marker = Makie.inherit(scene, :marker, :none),
+        camera_color = Makie.inherit(scene, :color, :black),
+        camera_markersize = Vec3f(2, 2, 3),
+        density = 30, # points per second
+        cycle = [:color,],
+    )
 end
+
+Makie.convert_arguments(::Type{<: PlotCameraPath}, path::Path, time::Number) = (path, Float64(time))
+Makie.convert_arguments(::Type{<: PlotCameraPath}, path::Path) = (path, 0.0)
+
+using Makie: Point3d
+function Makie.plot!(plot::PlotCameraPath)
+    eyepositions_obs = Observable{Vector{Point3d}}()
+    trange_obs = Observable{LinRange{Float64}}()
+    onany(plot, plot.path, plot.density; update = true) do path, density
+        tend = FlyThroughPaths.duration(path)
+        trange_obs.val = LinRange(0.0, Float64(tend), round(Int, tend*density))
+        eyepositions_obs.val = Makie.Point3d.(getproperty.(path.(trange_obs.val), :eyeposition))
+        notify(eyepositions_obs)
+        notify(trange_obs)
+    end
+    notify(plot.density) # run the `onany` once
+
+    eyeposition_obs = Observable{Point3d}(plot.path[](0).eyeposition)
+    lookat_obs = Observable{Point3d}(plot.path[](0).lookat)
+    viewdir_obs = Observable{Point3d}(plot.path[](0).lookat)
+
+    current_viewstate_obs = lift(plot, plot.path, plot.time) do path, time
+        current_viewstate = path(time)
+        eyeposition_obs.val = current_viewstate.eyeposition |> Point3d
+        lookat_obs.val = current_viewstate.lookat |> Point3d
+        viewdir = (current_viewstate.lookat - current_viewstate.eyeposition)
+        viewdir_obs.val = Point3d(viewdir)
```

Comment:
```suggestion
        eyeposition_obs.val = Point3d(current_viewstate.eyeposition)
        lookat_obs.val = Point3d(current_viewstate.lookat)
        viewdir = current_viewstate.lookat - current_viewstate.eyeposition
        viewdir_obs.val = Point3d(viewdir)
```
Just for consistency